modules/Bio/EnsEMBL/Compara/RunnableDB/Flatfiles/MySQLImportHomologies.pm

```diff
@@ -0,0 +1,164 @@
+Copyright [2016-2020] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+=pod
+
+=head1 NAME
+
+Bio::EnsEMBL::Compara::RunnableDB::Flatfiles::MySQLImportHomologies
+
+=head1 DESCRIPTION
+
+This runnable takes a homology dump file (and optionally, attribute files)
+and formats them to mysqlimport the data directly to the homology and
+homology_member tables.
+
+=cut
+
+package Bio::EnsEMBL::Compara::RunnableDB::Flatfiles::MySQLImportHomologies;
+
+use warnings;
+use strict;
+use Bio::EnsEMBL::Registry;
+use Bio::EnsEMBL::Compara::DBSQL::DBAdaptor;
+
+use Bio::EnsEMBL::Compara::Utils::FlatFile qw(map_row_to_header);
+
+use base ('Bio::EnsEMBL::Compara::RunnableDB::BaseRunnable');
+
+sub fetch_input {
+    my $self = shift;
+
+    # handle any attrib files that may have been passed
+    $self->param('attribs', {});
+    my $attrib_files = $self->param('attrib_files');
+    return unless ( $attrib_files );
+
+    # fetch all attributes from file list
+    my %attribs;
+    my $primary_key = 'homology_id';
+    foreach my $f ( @$attrib_files ) {
+        open( my $fh, '<', $f ) or die "Cannot open $f for reading";
+        my $header = <$fh>;
+        my @header_cols = split( /\s+/, $header );
+        die "No $primary_key found in $f - please check file header line\n" unless grep {$_ eq $primary_key} @header_cols;
+        while ( my $line = <$fh> ) {
+            my $row = map_row_to_header($line, \@header_cols);
+            my $primary_id = $row->{$primary_key};
+            delete $row->{$primary_key};
+            foreach my $attrib_name ( keys %$row ) {
+                $attribs{$primary_id}->{$attrib_name} = $row->{$attrib_name};
+            }
+        }
+        close $fh;
+    }
+
+    $self->param('attribs', \%attribs);
+}
+
+sub write_output {
+    my $self = shift;
+
+    my $filename = $self->param_required('homology_flatfile');
+    my $attribs  = $self->param('attribs');
+
+    # open homology file
+    open(my $hom_fh, '<', $filename) or die "Cannot open $filename for reading";
+    my $header_line = <$hom_fh>;
+    my @header_cols = split(/\s+/, $header_line);
+
+    # open output files - one per table in tmp space
+    my $homology_csv = $self->worker_temp_directory . '/homology.txt';
+    open(my $h_csv, '>', $homology_csv) or die "Cannot open $homology_csv for writing";
+    my $homology_member_csv = $self->worker_temp_directory . '/homology_member.txt';
```

Comment:
Just in case (since this has happened in the past and I cannot answer the question myself), is there a chance 2 jobs for this runnable will end up in the same node/worker and there will be a race condition in these files?

Reply:
`$self->worker_temp_directory` is unique for each worker, so I would say we're good